### PR TITLE
ICU: bump

### PR DESCRIPTION
**Pull request by extrowerk**
Bump, coexistence, updated patchset.

Warning: this build uses the ICU 61+ namespace. To change it to the 60 and earlier namespace consult the "readme.html" in the icu4c folder.

Patch doesn't applies cleanly, some part completly missing, so i kept only what useful.
The packaging stage have a bug somewhere along the devel package requires, help would be really appreciated.

### CODE REVIEW

**Comment by pulkomandy** [comment]
Don't change the namespace back to icu60, this is how ICU handles ABI versions. It makes sure all symbols are different from one version to the other, and as a result it is even possible to link a lib with an old ICU, another lib with a new one, and then use both libs in the same app without problems.

What's in the patches looks ok, but what happened to gcc2 support? Most of the old patch was about that, IIRC (as everything else, we can upstream to ICU).

General comments as I proofread this:
- It seems strange that we use separate data dirs for gcc2/gcc7, yet we use the same path for all ICU versions. How comes the files are interchangeable between ICU 5.4 and 6.2, but not between gcc2 and gcc7 versions?
- It seems strange that $libDir/icu ends up in the devel package. What's in there?

I'm ok to help but your description of the problem isn't very helpful. Do you have a buildlog or something?

**Comment by extrowerk** [comment]
The mentioned files in the patchset does not exist, i grepped the source tree to see if it is moved to somewhere else, but nothing.  Maybe i overlooked it.

The libdir/icu contains: "pkgdata.inc" and "Makefile.inc" files. I think they are correctly delegated to the devel package.

Packaging fixed in the meantime, but the test fails:

```
cnumtst.c: In function 'Test12052_NullPointer':
cnumtst.c:3070:34: error: 'u' undeclared (first use in this function)
     static const UChar input[] = u"199a";
                                  ^
cnumtst.c:3070:34: note: each undeclared identifier is reported only once for each function it appears in
cnumtst.c:3070:35: error: expected ',' or ';' before string constant
     static const UChar input[] = u"199a";
                                   ^~~~~~
```

**Comment by pulkomandy** [comment]
On Sun, Jul 22, 2018 at 07:44:55AM -0700, miqlas wrote:
> The mentioned files in the patchset does not exist, i grepped the source tree to see if it is moved to somewhere else, but nothing.  Maybe i overlooked it.

Yes, maybe the patches need to be re-done elsewhere. Or maybe they
changed their code slightly and it builds again with gcc2.

> The libdir/icu contains: "pkgdata.inc" and "Makefile.inc" files. I think they are correctly delegated to the devel package.

Yes, putting them in the devel package is right, but why are they in
$libDir?

> 
> Packaging fixed in the meantime, but the test fails:
> 
> ```
> cnumtst.c: In function 'Test12052_NullPointer':
> cnumtst.c:3070:34: error: 'u' undeclared (first use in this function)
>      static const UChar input[] = u"199a";

GCC2 does not support this notation. This may work:

    static const UChar input[] = {'1', '9', '9', 'a', 0};

-- 
Adrien.

**Comment by extrowerk** [comment]
I see this, maybe could be helpful:
"=== Important Notes: ===
Data Packaging: library
 This means: ICU data will be linked with ICU. A shared data library will be built. 
 To locate data: ICU will use the linked data library. If linked with the stub library located in stubdata/, the application can use udata_setCommonData() or set a data path to override."

In the meantime i found an another tarball wich contains all the required files, applied the patchset, looks good.

**Comment by extrowerk** [comment]
Tests needs some patching, but that is well beyon my skills.
Other than that it builds and packages fine.

**Comment by extrowerk** [comment]
Closes  #2598

File: dev-libs/icu/icu62-62.1.recipe [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+	devel:libicudata$secondaryArchSuffix = $libVersionCompat
+	devel:libicui18n$secondaryArchSuffix = $libVersionCompat
+	devel:libicuio$secondaryArchSuffix = $libVersionCompat
+	devel:libicutest$secondaryArchSuffix = $libVersionCompat
+	devel:libicutu$secondaryArchSuffix = $libVersionCompat
+	devel:libicuuc$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	icu62$secondaryArchSuffix == $portVersion base
+	"
```
Comment (korli) [inline dev-libs/icu/icu62-62.1.recipe]: A CONFLICTS_devel would be needed IMO.

**Comment by extrowerk** [comment]
It needs additional gcc2 patches, thanks @Begasus for testing it.

**Pull request closed without merging**